sequencer/src/state.rs

```diff
@@ -24,24 +32,30 @@ impl HotShotState for State {
 
     fn validate_and_apply_header(
         &self,
-        _proposed_header: &Self::BlockHeader,
-        _parent_header: &Self::BlockHeader,
+        proposed_header: &Self::BlockHeader,
+        parent_header: &Self::BlockHeader,
         _view_number: &Self::Time,
     ) -> Result<Self, Self::Error> {
-        todo!("Tracking issue: [https://github.com/EspressoSystems/espresso-sequencer/issues/968]")
+
+         // TODO validate fee state
+         if let Err(_) = _validate_proposal(parent_header, proposed_header) {
+             Err(BlockError::InvalidBlockHeader)
+         } else {
+             Ok(proposed_header.validated_state)
```

Comment:
It should now be possible to remove `validated_state` from the header, compute the updates to the state here using `self`, and simply verify the results against the merkle tree roots in `parent_header`/`proposed_header`

Reply:
182ffde should do as described.